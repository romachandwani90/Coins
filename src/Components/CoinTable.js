import { useEffect, useState } from "react";
import { getMarkets } from '../services/service';
import CoinMarketDetails from './CoinMarketDetails';

const CoinTable = () => {
    const [markets, setMarkets] = useState([]);
    useEffect(() => {
        const fetchMarkets = async () => {
            const result = await getMarkets();

            setMarkets(result);
        };
        if (!markets.length) {
            fetchMarkets();
        }
    });

    return <>
        {markets && <CoinMarketDetails key={markets.id} details={markets}/>}
    </>;
};

export default CoinTable;