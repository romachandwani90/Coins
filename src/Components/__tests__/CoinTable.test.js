import React from 'react'
import {render} from '@testing-library/react'
import CoinTable from '../CoinTable'
 
 it('should take a snapshot', () => {
    const { asFragment } = render(<CoinTable />)
    
    expect(asFragment(<CoinTable />)).toMatchSnapshot()
   })
