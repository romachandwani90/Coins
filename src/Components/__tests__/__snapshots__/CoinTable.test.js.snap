// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsrHT dtFYMQ"
  >
    <table>
      <thead>
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
          >
            Â 
          </th>
          <th
            colspan="2"
            role="columnheader"
          >
            Coin
          </th>
          <th
            colspan="3"
            role="columnheader"
          >
            Price Info
          </th>
        </tr>
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
          >
            #
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Name
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Symbol
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Current Price
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
             High 24-hour Price
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Low 24-hour Price
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
    <div
      class="pagination"
    >
      <button
        class="sc-bdnxRM jZAcTr"
        disabled=""
      >
        &lt;&lt;
      </button>
       
      <button
        class="sc-bdnxRM jZAcTr"
        disabled=""
      >
        &lt;
      </button>
       
      <button
        class="sc-bdnxRM jZAcTr"
        disabled=""
      >
        &gt;
      </button>
       
      <button
        class="sc-bdnxRM jZAcTr"
        disabled=""
      >
        &gt;&gt;
      </button>
       
      <span>
        Page 
        <strong>
          1 of 0
        </strong>
         
      </span>
      <span>
        | Go to page: 
        <input
          style="width: 100px; padding: 5px;"
          type="number"
          value="1"
        />
      </span>
       
      <select
        style="background: rgb(234, 236, 251); padding: 5px;"
      >
        <option
          value="10"
        >
          Show 10
        </option>
        <option
          value="20"
        >
          Show 20
        </option>
        <option
          value="30"
        >
          Show 30
        </option>
        <option
          value="40"
        >
          Show 40
        </option>
        <option
          value="50"
        >
          Show 50
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
